docker-compose-production.yml

```diff
@@ -0,0 +1,17 @@
+version: '3.8'
+services:
+  pl:
+    build:
+      context: .
+    image: prairielearn/prairielearn:local
+    ports:
+      - 3000:3000
+    volumes:
```

Comment:
Should we put the Postgres state on a volume so it isn't lost if the container is destroyed?

Reply:
I agree, this would be the best way to implement the database. How do you recommend we approach this as we need to populate the DB before adding it to the docker volume.

Reply:
Hmm, good question. We may just have to have folks manually create the DB and run the migrations. Alternatively, we could avoid using the version of Postgres built into the Docker container and use a separate [`postgres`](https://hub.docker.com/_/postgres) container, set up the DB and credentials via environment variables, and then have folks set `"runMigrations": true` in their config file.

Do you know how other Docker Compose setups handle this problem?

Reply:
I am currently doing research into this and will have to get back to you. From what I've seen so far, Docker Compose setups have the database in another container. 

I will try both implantations very soon and then we can discuss about the pros and cons of each method.

Reply:
@nwalters512 For some reason on windows, when adding a volume that points to the database on the docker container it gets permission errors. I think this has to do with the way windows mounts folders.

It took me a bit but I think I've got a solution that would work for all systems. We can use docker volumes and this would give us a persistent database. What do you think of this solution?

Reply:
Candidly, I don't think many folks are going to be running PrairieLearn in production on Windows machines, so I'm not particularly concerned with that use case, especially if it adds any complexity. If there's a very straightforward way to make it work though, let's check it out!

@firasm that works because we pre-populate the database when we build the `prairielearn/prairielearn` image. As soon as we try to mount something over the Postgres data directory, we'd lose that.

Reply:
The volume approach is the most straight forward and bulletproof in my opinion, no room for user error. 

You can test it out by changing your docker-file to look like this:
```sh
version: '3.8'
services:
  pl:
    build:
      context: .
    image: prairielearn/prairielearn:local
    ports:
      - 3000:3000
    volumes:
      - psql:/var/postgres
      - /var/run/docker.sock:/var/run/docker.sock
      - ${HOME}/pl_ag_jobs:/jobs
      
    container_name: pl
    environment:
      - HOST_JOBS_DIR=${HOME}/pl_ag_jobs
      - NODE_ENV=production
      
volumes:
   psql:
```

This will create a docker volume `psql` and use that in place of all the files in the database. A user can always backup a volume when they like.

Reply:
After doing further research I think it is best to let the user decide how they would like to have the database state as a volume on docker. I've added to the docs the way you could do this for both Linux/Mac and for Windows.